### PR TITLE
for #28: add metrics doc and final implementation

### PR DESCRIPTION
**Pull request by groovecoder**
See also: https://bugzilla.mozilla.org/show_bug.cgi?id=1288196

Supersedes https://github.com/mozilla/blok/pull/89 with:

* A branch from here on the mozilla repo (on which @clouserw is now a collaborator)
* https://github.com/mozilla/blok/commit/efa4b5cf3ca3344c899c3497eb5d646dbacdace7 which includes documentation for the "page-problem" telemetry payload

### CODE REVIEW

File: docs/metrics.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+```
+
+
+A Redshift schema for the payload:
+
+```lua
+local schema = {
+--   column name       field type   length  attributes   field name
+    {"originDomain",   "VARCHAR",   255,    nil,         "Fields[payload.originDomain]"},
+    {"trackerDomains", "VARCHAR",   1000,   nil,         "Fields[payload.trackerDomains]"},
```
Comment (groovecoder) [inline docs/metrics.md]: We will use this data to answer:

>     Which trackers are most commonly involved with breakage?

Will storing this as a `VARCHAR` make it hard to report against the individual tracking domains as a dimension?

Comment (rafrombrc) [inline docs/metrics.md]: Likely it will, yes, but AFAICT there's not much choice if we're using Redshift as the storage. :P

**Comment by rafrombrc** [comment]
I give this ping format a tentative r+ for now, with the caveat that we'll probably want to change it in the not-too-distant future to match the "event telemetry" format that's currently being developed in partnership with the UI telemetry efforts (see [draft proposal](https://docs.google.com/document/d/1hNuS9lUJMvMqgntZXbFA6xZBU9zBpQgo7x73-sXKRpI/edit#heading=h.5e74w45t1btr)).

**Comment by groovecoder** [comment]
Thanks for the comments and reviews. I plan to fix up the telemetry ping payloads after https://github.com/mozilla/blok/pull/109 lands.

But I may have found another (couple) ping(s) to send?

```
  {
    "originDomain": "www.redditmedia.com",
    "trackerDomains": ["ssl.google-analytics.com",
                       "z.moatads.com"],
    "breakage": "",
    "notes": "",
    "event": "blok-disabled"
  }
```
```
  {
    "originDomain": "www.redditmedia.com",
    "trackerDomains": ["ssl.google-analytics.com",
                       "z.moatads.com"],
    "breakage": "",
    "notes": "",
    "event": "blok-enabled"
  }
```

**Comment by groovecoder** [comment]
And https://github.com/mozilla/blok/issues/110 might help us build up a very valuable database of "unknown resource domains" - could we use Telemetry to store those too?

**Comment by groovecoder** [comment]
@rbillings or @pdehaan would you be willing to QA this ... ?

1. Open a tab to `about:config` and change `testpilot.env` to `dev`
2. Go to http://testpilot.dev.mozaws.net and install Test Pilot (if it's not already installed)
3. Go to http://testpilot.dev.mozaws.net/experiments/blok and enable Blok (I'm pointing the experiment at this branch's `.xpi` file)
4. Open a tab to `about:telemetry` and pick Ping data source: Archived ping data
5. Go to page(s) that triggers the blok page action (I've been using github.com)
6. Click around in the blok popup: Disable blok, enable blok, page works button, report problem button, submit breakage details
7. Go back to `about:telemetry` tab
  * [ ] You should see a number of new entries in the "Ping" drop-down like "YYYY/MM/DD HH:MMSS - testpilottest"
8. Select one or some of the entries
  * [ ] Below in the "Raw Payload" section, you should see the Blok payload with `"event": "blok-disabled|blok-enabled|page-works|page-problem|submit"`

**Comment by rbillings** [comment]
Verified the above. I'm not sure if it's related to this PR or not as I didn't test at all last week, but I noticed that the reported status ['you reported a problem' or 'you said page works well'] does not always display until after the user has disabled & re-enabled blok for the site. 
- was able to repro, but not 100% - I had trouble figuring out what would trigger this
- if it happens, it happens for all tabs
- tested only on Nightly
- opening new browser windows in same session sometimes works well and sometimes not - but if it doesn't work, it doesn't work at all until you disable/re-enable blok

**Comment by rbillings** [comment]
Testing on a new session I was able to get Blok reporting the submitted status on some tabs and not others. 
* I suspect this may be a time delay issue as sites that didn't report status, later did show it correctly without manipulating Blok
- there is no relationship between opening tabs vs new windows
- there is no pattern to the sites which submit immediately or not

**Comment by groovecoder** [comment]
I think that behavior is actually related to https://github.com/mozilla/blok/pull/109 and that it doesn't show the reported status until a page refresh.

Since it's not caused by the metrics pings, I'd like to merge this and work on that flaky behavior under https://github.com/mozilla/blok/issues/113, https://github.com/mozilla/blok/issues/115, and https://github.com/mozilla/blok/issues/116.

**Pull request merged**